### PR TITLE
fix(image_projection_based_fusion): multi class of one point

### PR DESCRIPTION
**Pull request by tzhong518**
## Description

- Enable assigning multiple class to one point, in order to be consistent with the training settings.
   - The value of each class is set as pow(2, class_index), e.g.  {"CAR": 1, "BICYCLE": 2, "PEDESTRIAN": 4}
   - The "CLASS" field of a point is the sum of all the classes, e.g. if the projected point is inside the "CAR" roi and "PEDESTRIAN" roi at the same time, its "CLASS" is 5.
   - The "CLASS“ value is decoded by voxel_generator into one-hot code, e.g 00000101, where each bit represents a class.

## Related links

https://tier4.atlassian.net/wiki/spaces/CT/pages/2851864699/SP+2023-08-08+PointPainting+Sync

## Tests performed

rosbag:https://drive.google.com/drive/folders/161wVB87jIu2E9CC9hRmaKwAAWjaDney_
image
The brightest points are assigned as pedestrian and car at the same time.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: perception/image_projection_based_fusion/src/pointpainting_fusion/node.cpp [hunk 1]
```diff
@@ -133,7 +133,7 @@ PointPaintingFusionNode::PointPaintingFusionNode(const rclcpp::NodeOptions & opt
     auto it = find(paint_class_names.begin(), paint_class_names.end(), cls);
     if (it != paint_class_names.end()) {
       int index = it - paint_class_names.begin();
-      class_index_[cls] = index + 1;
+      class_index_[cls] = pow(2, index);
```
Comment (yukkysaito) [inline perception/image_projection_based_fusion/src/pointpainting_fusion/node.cpp:136]: The code is tricky and difficult to understand. Could you please comment or refactor it?

File: perception/image_projection_based_fusion/src/pointpainting_fusion/voxel_generator.cpp [hunk 2]
```diff
@@ -60,7 +60,8 @@ std::size_t VoxelGenerator::pointsToVoxels(
       point[2] = point_current.z();
       point[3] = time_lag;
       for (std::size_t i = 1; i <= config_.class_size_; i++) {
-        point[3 + i] = (*class_iter == i) ? 1 : 0;
+        auto decode = std::bitset<8>(*class_iter).to_string();
+        point[3 + i] = decode[-i] == 1 ? 1 : 0;
```
Comment (yukkysaito) [inline perception/image_projection_based_fusion/src/pointpainting_fusion/voxel_generator.cpp:64]: Same comment
https://github.com/autowarefoundation/autoware.universe/pull/4705/files#r1307026629

File: perception/image_projection_based_fusion/src/pointpainting_fusion/node.cpp (hunk 1 above)
Comment (tzhong518) [reply perception/image_projection_based_fusion/src/pointpainting_fusion/node.cpp]: I added in description and comment https://github.com/autowarefoundation/autoware.universe/pull/4705/commits/b730594e289fccfab5936eb5547faeee4dfee7e7#diff-e8df78707e8cee5e869c96c349aca84fa2126bc70e25f093bf10f8a325e638e2R136

File: perception/image_projection_based_fusion/src/pointpainting_fusion/voxel_generator.cpp (hunk 2 above)
Comment (tzhong518) [reply perception/image_projection_based_fusion/src/pointpainting_fusion/voxel_generator.cpp:65]: Added a comment https://github.com/autowarefoundation/autoware.universe/pull/4705/commits/b730594e289fccfab5936eb5547faeee4dfee7e7#diff-e8df78707e8cee5e869c96c349aca84fa2126bc70e25f093bf10f8a325e638e2R136

**Comment by yukkysaito** [comment]
@yukke42 can I ask for a review?

**Review by yukke42 — APPROVED** [review]
LGTM

**Pull request merged**